### DISCUSSION TITLE
What is the best practice for loading a global state variable on client-side?

### DISCUSSION THREAD

**Discussion by l33oo** [nuxt/nuxt#25551] [Questions] [answered]
I have a very basic question, but so far I haven't found the answer anywhere. How can I asynchronously load a global state variable on the client side and avoid hydration errors?

I've tried using middleware, plugin setup, and even the plugin hook `app:mounted`, but none of them seem to be suitable.

Here's a sandbox I created for reference: [sandbox link](https://codesandbox.io/p/devbox/unruffled-jang-8f7ywm?file=%2Fplugins%2Ftest.client.ts%3A18%2C1). If loaded correctly, you should see a red `red` turning into a black `black`. However, in reality, it appears as a red `black`.

Could you please help me with this? Thank you!

**Comment by xjccc** [comment]
use `page:finish` have a try?

**Comment by TheAlexLichter** [comment, marked as answer]
In `app.vue` or in the layout @ `onBeforeMount` or `onMounted`. [Example](https://stackblitz.com/edit/nuxt-discussions-25551?). The `onNuxtReady` hook would work too but comes in later ☺️

**Reply by MarioC3** [reply]
Would this be equivalent to fetching the data on a client-only plugin?

**Reply by TheAlexLichter** [reply]
A client-only plugin would run *before* `App.vue`
